src/StructUtils.hs

```diff
@@ -1,19 +1,33 @@
 module StructUtils where
 
-import Managed
+import Interfaces
 import Obj
 import Polymorphism
 import Types
 
-memberInfo :: TypeEnv -> Ty -> (Ty, String, Ty)
-memberInfo typeEnv memberTy =
-  let refOrNotRefType = if isManaged typeEnv memberTy then RefTy memberTy (VarTy "w") else memberTy -- OBS! The VarTy "w" here is dubious
-   in (refOrNotRefType, if isManaged typeEnv memberTy then "&" else "", FuncTy [refOrNotRefType] StringTy StaticLifetimeTy)
+-- | The 'str'/'prn' functions for primitive types don't take refs, while other types do
+-- so we need to adjust for that when finding and calling them in compound types.
+-- The returned tuple contains ("" || "&", `str function type`).
+memberStrCallingConvention :: String -> TypeEnv -> Env -> Ty -> (String, Ty)
+memberStrCallingConvention strOrPrn typeEnv globalEnv memberTy =
+  if callWithValue
+    then ("", withValueSig)
+    else ("&", withRefSig)
+  where
+    withValueSig = FuncTy [memberTy] StringTy StaticLifetimeTy
+    withRefSig = FuncTy [RefTy memberTy (VarTy "w")] StringTy StaticLifetimeTy -- "w" here is dubious?
```

Comment:
Yes, I think using `w` is potentially problematic. We likely need a function here that ensures we don't create a collision w/ the ty vars generated/rewritten by the other parts of the compiler?

Reply:
Yeah, usually there's an XObj around that has a piece of unique information we can use for this. 

I'll try to make this function take such an `Info` as an additional argument.

Reply:
Hmm, it was harder than expected. The current solution might not actually be a problem since all type varaibles are uniquified before being solved (at the call site). Still, since I might have missed some detail I'll leave the comment in here for now.